src/MetaMorpho.sol

```diff
@@ -375,7 +375,7 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
             uint256 withdrawn = supplyAssets.zeroFloorSub(allocation.assets);
 
             if (withdrawn > 0) {
-                if (allocation.marketParams.loanToken != asset()) revert ErrorsLib.InconsistentAsset(id);
+                if (!config[id].enabled) revert ErrorsLib.MarketNotEnabled(id);
```

Comment:
should we move it outside of the `if`?

Reply:
No because the supply cap is checked upon supply and `supplyCap > 0 => enabled == true`

Reply:
yes ofc, but we could have reverted earlier, even if i know we don't really care

Reply:
In the case of the supply we don't want to revert, we want to continue

Reply:
indeed